### PR TITLE
fix: [PL-38956]: show RBAC messages on Secrets Page

### PR DESCRIPTION
**Pull request by namanharness**
### Summary
Fixed Bugs -

1.)Delegate Tags shown in irregular manner. [Link](https://harness.atlassian.net/browse/PL-42822)

| Before | After |
|--------|--------|
| image | image | 

2.) Secrets Page does not show RBAC Message. 

| Before | After |
|--------|--------|
| image | image | 

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/27-platform/secrets/pages/secrets/SecretsPage.tsx [hunk 1]
```diff
@@ -249,6 +249,9 @@ const SecretsPage: React.FC<SecretsPageProps> = ({ mock }) => {
 
       <Page.Body
         className={css.body}
+        loading={loading}
+        error={error ? getRBACErrorMessage(error) : ''}
+        retryOnError={() => refetch()}
```
Comment (ishantmehtaharness) [inline src/modules/27-platform/secrets/pages/secrets/SecretsPage.tsx:254]: We can change this to `retryOnError={refetch}`
We dont need to create a function here

**Review by ishantmehtaharness — COMMENTED** [review]
Added comments

File: src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx [hunk 2]
```diff
@@ -81,7 +81,7 @@ const RenderTags: Renderer<CellProps<DelegateGroupDetails>> = ({ row }) => {
   return Object.keys(delegateTags).length > 0 ? (
     <>
       <Text lineClamp={1} margin={{ right: 'medium' }}>
-        <TagsRenderer tags={delegateTags} length={3} />
+        <TagsRenderer tags={delegateTags} length={1} />
```
[the remark this answers is not in the document]
Comment (namanharness) [reply src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:84]: yes so when there are more than 1 it crumbles up , creates 2 pop ups
one to show the text which is overflowing and second one to show remaining tags

So i think one would be ideal, for long names and rest in the tags pop up

File: src/modules/27-platform/secrets/pages/secrets/SecretsPage.tsx (hunk 1 above)
Comment (namanharness) [reply src/modules/27-platform/secrets/pages/secrets/SecretsPage.tsx:254]: image

Getting this when trying to add suggested code.

Comment (ishantmehtaharness) [reply src/modules/27-platform/secrets/pages/secrets/SecretsPage.tsx:254]: I see. Refetch wants to be called without any arguments 
Looks good in that case

**Comment by ishantmehtaharness** [comment]
Looks good

**Review by ishantmehtaharness — APPROVED** [review]
Looks good to me

File: src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx (hunk 2 above)
Comment (mayankVermaHarness) [inline src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:84]: What happens when a single tag itself has a very long name?

Comment (namanharness) [reply src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:84]: image

Behaviour would be same like this ⬆️ 
some portion of the name will be shown and on hover a pop up with complete name and + icon with another popup showing remaining tags.

Any better solution to this?

Comment (avinashmadhwani02) [reply src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:84]: Should we add some padding in the first tooltip? 
+4 is not looking good without padding

**Review by avinashmadhwani02 — APPROVED** [review]

**Comment by namanharness** [comment]
https://github.com/harness/harness-core-ui/pull/19026#discussion_r1400102685
sure @avinashmadhwani02 , i am assuming this is from UI core, i will look into it.

**Pull request merged**